src/main/daml/Daml/Finance/Account/Account.daml

```diff
@@ -33,29 +38,56 @@ template Account
     observers : PartiesMap
       -- ^ Observers.
   where
-    signatory custodian, owner
+    signatory custodian, owner, Lockable.getLockers this
     observer Disclosure.flattenObservers observers
 
     -- Outgoing controllers must be non-empty.
-    ensure not . S.null $ controllers.outgoing
+    ensure isValidLock lock && (not . S.null $ controllers.outgoing)
 
-    let account = AccountKey with custodian; owner; id
+    let
+      account = AccountKey with custodian; owner; id
+      mustNotBeLocked = assertMsg @Update "Must not be locked" $ isNone lock
 
     interface instance Account.I for Account where
       view = Account.View with custodian; id; owner; holdingFactoryCid; description; controllers
       getKey = account
-      credit Account.Credit{quantity} =
+      credit Account.Credit{quantity} = do
+        mustNotBeLocked
         exercise holdingFactoryCid HoldingFactory.Create with
           instrument = quantity.unit
           account
           amount = quantity.amount
           observers = M.empty
       debit Account.Debit{holdingCid} = do
-        vHolding <- view <$> fetch holdingCid
-        assertMsg "accounts must match" $ vHolding.account == account
+        mustNotBeLocked
+        holding <- fetch holdingCid
+        let
+          vHolding = view holding
+          vLockable = view $ toInterface @Lockable.I holding
+        assertMsg "Holding must not be locked" $ isNone vLockable.lock
+        assertMsg "Accounts must match" $ vHolding.account == account
         exercise holdingFactoryCid HoldingFactory.Remove with
           actors = S.fromList [account.custodian, account.owner]; holdingCid
 
+    interface instance Lockable.I for Account where
+      view = Lockable.View with lock; controllers = S.singleton custodian
+      acquire Lockable.Acquire{newLockers; context; lockType} = do
+        newLockableCid <- toInterfaceContractId @Disclosure.I <$>
+          acquireImpl this.lock (\lock -> this with lock) Lockable.Acquire{newLockers; context; lockType}
```

Comment:
I believe you can do the following to avoid the deconstruction and reconstruction of the `Lockable.Acquire` :

```suggestion
      acquire lockable@Lockable.Acquire{newLockers; context; lockType} = do
        newLockableCid <- toInterfaceContractId @Disclosure.I <$>
          acquireImpl this.lock (\lock -> this with lock) lockable
```